src/djc_ext_pydantic/monkeypatch.py

```diff
@@ -0,0 +1,50 @@
+from typing import Any, Type
+
+from django.template import NodeList
+from django.utils.safestring import SafeString
+from django_components import Component, Slot, SlotFunc
+from pydantic_core import core_schema
+
+
+# For custom classes, Pydantic needs to be told how to handle them.
+# We achieve that by setting the `__get_pydantic_core_schema__` attribute on the classes.
+def monkeypatch_pydantic_core_schema() -> None:
```

Comment:
This file modifies the classes defined in Django components / Django, so that they these classes can be used in typing.